android/java/org/chromium/chrome/browser/settings/BraveWalletAddNetworksFragment.java

```diff
@@ -72,227 +91,252 @@ public void onDestroy() {
+     *   <li>RPC URL, icon URL and block explorer URL support only HTTP and HTTPS protocols.
+     *   <li>Chain ID, and RPC URL are cross validated: if the chain ID returned by RPC endpoint
+     *       doesn't match the chain ID provided an error message will be displayed.
+     * </ul>
+     */
+    private void validateInputsAddChain() {
+        boolean error = false;
+        mSubmitError.setVisibility(View.INVISIBLE);
+
         NetworkInfo chain = new NetworkInfo();
         String strChainId = mChainIdEditText.getText().toString().trim();
         try {
-            int iChainId = Integer.valueOf(strChainId);
-            if (iChainId <= 0) {
+            int parsedChainId = Integer.parseInt(strChainId);
+            if (parsedChainId <= 0) {
                 mChainIdEditText.setError(
                         getString(R.string.brave_wallet_add_network_chain_id_error));
-
-                return false;
+                error = true;
+            } else {
+                strChainId = "0x" + Integer.toHexString(parsedChainId);
+                chain.chainId = strChainId;
+                chain.supportedKeyrings = new int[0];
             }
-            strChainId = "0x" + Integer.toHexString(iChainId);
         } catch (NumberFormatException exc) {
-            mChainIdEditText.setError(getString(R.string.brave_wallet_add_network_chain_id_error));
-
-            return false;
+            error = true;
+            PostTask.postTask(
+                    TaskTraits.UI_DEFAULT,
+                    () ->
+                            mChainIdEditText.setError(
+                                    getString(R.string.brave_wallet_add_network_chain_id_error)));
         }
-        chain.chainId = strChainId;
-        chain.supportedKeyrings = new int[0];
 
-        EditText chainName = view.findViewById(R.id.chain_name);
-        String strChainName = chainName.getText().toString().trim();
+        String strChainName = mChainName.getText().toString().trim();
         if (strChainName.isEmpty()) {
-            chainName.setError(getString(R.string.brave_wallet_add_network_chain_empty_error));
-
-            return false;
+            mChainName.setError(getString(R.string.brave_wallet_add_network_chain_empty_error));
+            error = true;
+        } else {
+            chain.chainName = strChainName;
         }
-        chain.chainName = strChainName;
 
-        EditText chainCurrencyName = view.findViewById(R.id.chain_currency_name);
-        String strChainCurrencyName = chainCurrencyName.getText().toString().trim();
-        if (strChainCurrencyName.isEmpty()) {
-            chainCurrencyName.setError(
-                    getString(R.string.brave_wallet_add_network_chain_empty_error));
+        chain.symbolName = mChainCurrencyName.getText().toString().trim();
 
-            return false;
-        }
-        chain.symbolName = strChainCurrencyName;
+        chain.symbol = mChainCurrencySymbol.getText().toString().trim();
```

Comment:
the same question as above about the name

Reply:
I would make the field of currency symbol mandatory as well.